### ISSUE TITLE
https://acpica.org/downloads - both links to tarballs are broken

### ISSUE THREAD

**Issue by kanavin** [open-acpica/acpica#810]
Seems like ...tar_0.gz needs to be renamed to ...tar.gz:

https://acpica.org/node/201

**Comment by ElyesH** [comment]
Any update on current issue ?
The link to acpica-unix-20221022.tar.gz is broken .

**Comment by ElyesH** [comment]
here is the broken link
(https://acpica.org/sites/acpica/files/acpica-unix-20221022.tar.gz)

**Comment by acpibob** [comment]
This has come up before, it is the result of attempting to upload the file, which then is too big for the website.  I can point the link to the correct filename, but I don't know how to actually rename the file.

**Comment by kanavin** [comment]
How about hosting the tarballs directly here on github?
https://github.com/acpica/acpica/releases

**Comment by acpibob** [comment]
I think we want to host the tarballs directly on acpica.org for simplicity.

**Comment by kanavin** [comment]
But can we make progress somehow? Can you respin the release and upload completely new, correctly named tarballs?

**Comment by ElyesH** [comment]
> But can we make progress somehow? Can you respin the release and upload completely new, correctly named tarballs?

Would this  https://github.com/acpica/acpica/releases/tag/R10_20_22  be a workaround ?

**Comment by kanavin** [comment]
> Would this https://github.com/acpica/acpica/releases/tag/R10_20_22 be a workaround ?

Unfortonately not. These tarballs are auto-generated by github, and their content may change over time (different compression settings or order of files etc), meaning their integrity can't be reliably verified against an external checksum.

Also it would help if acpica tags would sort properly, e.g. year ahead of month and day. Then we (the yocto project) would just clone from git (and run automated checks to determine the latest tag) instead of trying to use tarballs.

**Comment by chenrui333** [comment]
> > Would this https://github.com/acpica/acpica/releases/tag/R10_20_22 be a workaround ?
> 
> Unfortonately not. These tarballs are auto-generated by github, and their content may change over time (different compression settings or order of files etc), meaning their integrity can't be reliably verified against an external checksum.

I dont think so, the tarball/checksum is actually always preferred thing used by homebrew-core project.

**Comment by kanavin** [comment]
> IMO tarball source archive is the same deal as git checkout assuming they are referring to the same git commit. Besides, git checkout also needs to consider the git submodules situation.

The tarballs auto-generated by github from tags are not stable. They can, and do change over time. So they cannot be verified against a checksum, because the checksum may change without warning, which makes you vulnerable to supply chain attacks. If you use a tarball, it needs to never change.

I see you used it here:
https://github.com/Homebrew/homebrew-core/pull/120652/files
What are you going to do when the checksum changes and builds start failing? Can you be confident that it's not a supply chain attack, and do you really want to update those checksums regularly?

**Comment by kanavin** [comment]
Here's a recent, in-depth look on github's auto-archives going wrong. 

We'd all do well to avoid both the problem and the fights around it by not using them:
https://lwn.net/Articles/921787/

**Comment by maciejwieczorretman** [comment]
I believe you can now download release files from github like:
```bash
wget https://github.com/open-acpica/acpica/releases/download/20260408/acpibin
```

**Comment by andy-shev** [comment]
Same questions as in #883.
